tests-integration/regressions_test.go

```diff
@@ -228,3 +229,119 @@ func TestMalformedEventsState(t *testing.T) {
 			},
 		}))
 }
+
+// Regression test for https://github.com/matrix-org/sliding-sync/issues/295
+// This test:
+// - injects a good room and a bad room in the v2 response
+// - then injects a good room update if the since token advanced
+// - checks we see all updates
+// In the past, the bad room in the first v2 response caused the proxy to retry and never advance,
+// wedging the poller.
+func TestBadCreateInitialiseDoesntWedgePolling(t *testing.T) {
+	pqString := testutils.PrepareDBConnectionString()
+	// setup code
+	v2 := runTestV2Server(t)
+	v3 := runTestServer(t, v2, pqString)
+	defer v2.close()
+	defer v3.close()
+
+	goodRoom := "!good:localhost"
+	badRoom := "!bad:localhost"
+
+	v2.addAccount(t, alice, aliceToken)
+	// we should see the since token increment, if we see repeats it means
+	// we aren't returning DataErrors when we should be.
+	wantSinces := []string{"", "1", "2"}
+	ch := make(chan bool)
```

Comment:
To check: we don't use the bool here; we're just using the channel a something that we can `select` over and close to indicate completion?

Reply:
Yes